Tests/SwiftAlgorithmsTests/TrimTests.swift

```diff
@@ -56,4 +56,14 @@ final class TrimTests: XCTestCase {
     let results_one = [2, 10, 12, 15, 20, 100].trimming { $0.isMultiple(of: 2) }
     XCTAssertEqual(results_one, [15])
   }
+    
+  func testTrimmingPrefix() {
+    let results = [2, 10, 12, 15, 20, 100].trimmingPrefix { $0.isMultiple(of: 2) }
+    XCTAssertEqual(results, [15, 20, 100])
+  }
+
+  func testTrimmingSuffix() {
+    let results = [2, 10, 12, 15, 20, 100].trimmingSuffix { $0.isMultiple(of: 2) }
+    XCTAssertEqual(results, [2, 10, 12, 15])
+  }
```

Comment:
We'll also need tests for the new mutating methods. Can you include a test that uses `ArraySlice` as a type to verify that the ambiguity from above is resolved? For example:

```swift
var values = [2, 10, 12, 15, 20, 100] as ArraySlice
values.trim(while: { $0.isMultiple(of: 2) })
```